driver-core/src/test/functional/com/mongodb/client/model/expressions/DateExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+
+import static com.mongodb.client.model.expressions.Expressions.of;
+
+class DateExpressionsFunctionalTest extends AbstractExpressionsFunctionalTest {
```

Comment:
We can use the Java SE API to compute expected values: https://github.com/stIncMale/mongo-java-driver/commit/e087e7fb9a334cb176024e9abcb0044b23d6fb74.

Reply:
cherrypicked